### PR TITLE
machineconfig: Use nosmt KernelArgument now

### PR DESCRIPTION
**Pull request by cgwalters**
This replaces https://github.com/openshift/installer/pull/2805

Eventually it'd be nice to drop the deprecated `/etc/pivot/`
stuff from the MCO, this is a step towards that.

### CODE REVIEW

**Comment by cgwalters** [comment]
(No bug, this can wait for 4.7)

File: pkg/asset/machines/machineconfig/hyperthreading.go [hunk 1]
```diff
@@ -41,7 +22,8 @@ func ForHyperthreadingDisabled(role string) (*mcfgv1.MachineConfig, error) {
 			},
 		},
 		Spec: mcfgv1.MachineConfigSpec{
-			Config: rawExt,
```
Comment (abhinavdahiya) [inline pkg/asset/machines/machineconfig/hyperthreading.go:44]: https://github.com/openshift/machine-config-operator/blob/master/install/0000_80_machine-config-operator_01_machineconfig.crd.yaml#L70

That makes it feel like ignition field is required

Comment (cgwalters) [reply pkg/asset/machines/machineconfig/hyperthreading.go:44]: Hmm possibly, I didn't test this locally; we also don't have CI coverage for this which probably needs to happen as a prerequisite.   (It is IMO kind of a bug if the ignition section is really required becuase in exactly this case we're just using the "high level" MC section and not Ignition)

Comment (LorbusChris) [reply pkg/asset/machines/machineconfig/hyperthreading.go:44]: As of today, at least this has to be passed:
```
ignConfig := igntypes.Config{
    Ignition: igntypes.Ignition{
	Version: igntypes.MaxVersion.String(),
    },
}
```

**Pull request closed without merging**